nextflow_schema.json

```
@@ -43,37 +64,174 @@
+                    "description": "File in FASTA format containing possible adapters to remove. Accepted formats: *.{fasta,fna,fas,fa}"
+                },
+                "save_trimmed_fail": {
+                    "type": "boolean",
+                    "description": "Specify true to save files that failed to pass trimming thresholds ending in *.fail.fastq.gz"
+                },
+                "save_merged": {
+                    "type": "boolean",
+                    "description": "Specify true to save all merged reads to the a file ending in *.merged.fastq.gz"
+                },
+                "skip_fastp": {
+                    "type": "boolean",
+                    "description": "Skip the fastp process if true"
+                },
+                "skip_fastqc": {
+                    "type": "boolean",
+                    "description": "Skip FastQC processes if true"
+                },
+                "extra_fastp_args": {
+                    "type": "string",
+                    "description": "Extra arguements for fastp. For example, '--trim_front1 15 --trim_front2 15 --trim_tail1 5 --trim_tail2 5'"
+                }
+            },
+            "fa_icon": "fas fa-cut"
+        },
+        "sortmerna_options": {
+            "title": "SortMeRNA options",
+            "type": "object",
+            "description": "",
+            "default": "",
+            "properties": {
+                "remove_ribo_rna": {
+                    "type": "boolean",
+                    "description": "Enable the removal of reads derived from ribosomal RNA using SortMeRNA."
+                },
+                "ribo_database_manifest": {
+                    "type": "string",
+                    "default": "${projectDir}/assets/rrna-db-defaults.txt",
+                    "description": "Text file containing paths to fasta files (one per line) that will be used to create the database for SortMeRNA."
+                },
+                "save_non_ribo_reads": {
+                    "type": "boolean",
+                    "description": "If this option is specified, intermediate FastQ files containing non-rRNA reads will be saved in the results directory."
+                }
+            },
+            "fa_icon": "fas fa-filter"
+        },
+        "assembly_options": {
+            "title": "Assembly options",
+            "type": "object",
+            "description": "",
+            "default": "",
+            "properties": {
+                "assemblers": {
+                    "type": "string",
+                    "default": "trinity,rnaspades",
+                    "description": "Assemblers to use. Possible options include \"trinity\", \"trinity_no_norm\" (trinity without normalized reads), and \"rnaspades\"."
```

```suggestion
                    "description": "Assemblers to use. Possible options include `trinity`, `trinity_no_norm` (trinity without normalized reads), and `rnaspades`."
```
you could add some automatic validation of the correct assembles by providing a regex pattern in the `pattern` field for this parameter